perimeter/shared_access.sp

```diff
@@ -895,3 +908,897 @@ control "rds_db_snapshot_shared_with_trusted_accounts" {
     service = "AWS/RDS"
   })
 }
+
+benchmark "resource_policy_shared_access" {
+  title         = "Shared Access"                                                                                                                          # TODO
+  description   = "Resources should only be shared with trusted entities through AWS Resource Access Manager (RAM), configurations, or resource policies." # TODO
+  documentation = file("./perimeter/docs/resource_policy_shared_access.md")
+  children = [
+    benchmark.resource_policy_shared_accounts_access,
+    benchmark.resource_policy_shared_organizations_access,
+    benchmark.resource_policy_shared_services_access,
+    benchmark.resource_policy_shared_identity_providers_access,
+  ]
+
+  tags = merge(local.aws_perimeter_common_tags, {
+    type = "Benchmark"
+  })
+}
+
+locals {
+  resource_policy_shared_account_sql = <<-EOT
+    select
+      r.__ARN_COLUMN__ as resource,
+      case
+        when jsonb_array_length(pa.allowed_principal_account_ids) = 0 then 'ok'
+        when pa.access_level = 'private' then 'ok'
+        when pa.allowed_principal_account_ids <@ to_jsonb(($1)::text[]) then 'ok'
+        else 'alarm'
+      end as status,
+      case
+        when jsonb_array_length(pa.allowed_principal_account_ids) = 0 then title || ' trust policy does not reference any accounts.'
+        when pa.access_level = 'private' then title || ' trust policy does not reference any cross-accounts.'
+        when pa.allowed_principal_account_ids <@ to_jsonb(($1)::text[]) then concat(
+          title, 
+          ' trust policy grants cross-account access to ',
+          jsonb_array_length(pa.allowed_principal_account_ids),
+          ' trusted account(s).'
+        )
+        when jsonb_array_length(to_jsonb(pa.allowed_principal_account_ids) - ($1)::text[]) = 1 then concat(
```

Comment:
Similar to the suggestion above for the public access `alarm` case, if we changed all of the `alarm` cases to use the following format:
```sql
else title || ' policy grants access to ' || jsonb_array_length(pa.public_statement_ids) || ' untrusted account(s): [' || pa. allowed_principal_account_ids || '].'
```

So some sample reasons would look like:
- `s3-bucket-test policy grants access to 1 untrusted account(s): [0123456789012].`
- `s3-bucket-test policy grants access to 3 untrusted account(s): [0123456789012, 0123456789013, 0123456789014].`
- `s3-bucket-test policy grants access to 10 untrusted account(s): [0123456789012, 0123456789012, 0123456789013, 0123456789014, 0123456789015, 0123456789016, 0123456789017, 0123456789018, 0123456789019, 0123456789020].`

Does the console + dashboard Ui output render this without any errors? I'm expecting it to get cut off if the list of items is longer than 2 or 3, but I think this may be ok since our current set of reasons only show this many anyway.

Also, I believe users can output to CSV, JSON, or another format, in order to get the full reason (need to verify this though).